Dockerfile

```diff
@@ -1,32 +1,35 @@
 FROM mhart/alpine-node:6
 LABEL maintainer "Kyle Holzinger <kylelholzinger@gmail.com>"
 
-WORKDIR /usr/src/app
+RUN adduser -u 9000 -D app
 
-# For cache purpose
-ARG VER_TSLINT=5.0.0
+WORKDIR /usr/src/app
 
-COPY engine.json ./
+COPY engine.json package.json yarn.lock ./
 COPY ./bin/ ./bin/
 
-RUN npm install --global yarn && \
+RUN \
+  : Install needed tools && \
+  npm install --global yarn && \
   apk --update add git jq && \
-  bin/get-tslint-rules "$VER_TSLINT" && \
-  cat engine.json | jq ".version = \"$VER_TSLINT\"" > /tmp/engine.json && \
+  : Install NPM packages && \
```

Comment:
Do we need these comments? The commands seem fairly revealing on their own.

Reply:
No, not really. I added them to the original chain to annotate it. It was a bit more cryptic.